### PR TITLE
Accurate NumericIdMapper hints

### PR DESCRIPTION
**Pull request by klaussfreire**
Implement accurate hints for NumericIdMapper and derivatives,
finally getting improved access patterns for all id mappers.

### CODE REVIEW

File: doc/source/CHANGELOG.rst [hunk 1]
```diff
@@ -5,6 +5,15 @@ All notable changes to this project will be documented here.
 
 The format is largely inspired by keepachangelog_.
 
+v0.8.2 - Unreleased
```
Comment (tzulberti-jampp) [inline doc/source/CHANGELOG.rst:8]: Why patch the patch version if this isn't a bugfix. Shouldn't we patch de minor?

**Review by llogiudice — APPROVED** [review]

File: doc/source/CHANGELOG.rst (hunk 1 above)
Comment (klaussfreire) [reply doc/source/CHANGELOG.rst:8]: It's completely transparent and backward compatible as it's only performance. The performance issue could be considered a bug if you squint a bit.

Comment (klaussfreire) [reply doc/source/CHANGELOG.rst:8]: It also avoids having to patch FaultClient

Comment (tzulberti-jampp) [reply doc/source/CHANGELOG.rst:8]: I believe this is the real reason...

**Review by tzulberti-jampp — APPROVED** [review]

**Pull request merged**